sdk/search/Azure.Search.Documents/src/Indexes/VectorSearchFieldAttribute.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Search.Documents.Indexes.Models;
+
+namespace Azure.Search.Documents.Indexes
+{
+    /// <summary>
+    /// Attributes a vector field using a collection of single type.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
+    public class VectorSearchFieldAttribute : Attribute, ISearchFieldAttribute
+    {
+        /// <summary>
+        /// The dimensionality of the vector field.
+        /// </summary>
+        public int VectorSearchDimensions { get; set; } = 0;
+
+        /// <summary>
+        /// The name of the vector search algorithm configuration that specifies the algorithm and optional parameters for searching the vector field.
+        /// </summary>
+        public string VectorSearchProfileName { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether the field is returned in search results. The default is false.
+        /// </summary>
+        public bool IsHidden { get; set; }
+
+        /// <inheritdoc/>
+        void ISearchFieldAttribute.SetField(SearchField field) => SetField(field);
+
+        private protected void SetField(SearchField field)
+        {
+            field.IsSearchable = true;
+            field.IsHidden = IsHidden;
+
+            if (VectorSearchDimensions != 0)
```

Comment:
Few questions, does field builder need to do any validation here? Both dimensions and the profile name are required for the SearchField to be a valid vectorizable field and dimensions would need to be > 0

Reply:
I have removed the default value for `VectorSearchDimensions` and this condition, which I believe is unnecessary now that we have a separate type for vector fields. I also believe there is no need for validation here, as the service will handle all validations and throw the following error:
>The vector field 'FieldName' must have the property 'dimensions' set to a positive, non-zero integer between 2 and 2048.